ui/common/src/dialog.ts

```diff
@@ -1,11 +1,54 @@
-import { onInsert, looseH as h, VNode } from 'common/snabbdom';
-import { isTouchDevice } from 'common/device';
-import * as xhr from 'common/xhr';
-import * as licon from 'common/licon';
+import { onInsert, looseH as h, VNode, Attrs, LooseVNodes } from './snabbdom';
+import { isTouchDevice } from './device';
+import * as xhr from './xhr';
+import * as licon from './licon';
 
 let dialogPolyfill: { registerDialog: (dialog: HTMLDialogElement) => void };
 
-// for usage see file://./../../../@types/lichess/dialog.d.ts
+export interface Dialog {
+  readonly open: boolean; // is visible?
+  readonly view: HTMLElement; // your content div
+  readonly returnValue?: 'ok' | 'cancel' | string; // how did we close?
+
+  showModal(): Promise<Dialog>; // resolves on close
+  show(): Promise<Dialog>; // resolves on close
+  actions(actions?: Action | Action[]): void; // set new or reattach existing actions
+  close(): void;
+}
+
+export interface DialogOpts {
+  class?: string; // zero or more classes for your view div
+  css?: ({ url: string } | { hashed: string })[]; // fetches hashed or full url css
+  htmlText?: string; // content, text will be used as-is
+  cash?: Cash; // content, overrides htmlText, will be cloned and any 'none' class removed
+  htmlUrl?: string; // content, overrides htmlText and cash, url will be xhr'd
+  append?: { node: HTMLElement; where?: string; how?: 'after' | 'before' | 'child' }[]; // default 'child'
```

Comment:
this new `append.how` doesn't seem to be used anywhere. It still compiles after I remove it.